dotcom-rendering/src/lib/useAdBlockAsk.ts

```diff
@@ -69,18 +70,36 @@ const detectByRequests = async () => {
 	return !doubleclickSuccess && guardianSuccess;
 };
 
-export const useAdblockAsk = (slotId: string): boolean => {
+export const useAdblockAsk = ({
+	slotId,
+	shouldHideReaderRevenue,
+	isPaidContent,
+}: {
+	slotId: `dfp-ad--${string}`;
+	shouldHideReaderRevenue: boolean;
+	isPaidContent: boolean;
+}): boolean => {
 	const isInVariant = useIsInAdBlockAskVariant();
 	const [adBlockerDetected, setAdBlockerDetected] = useState<boolean>(false);
 
+	const authStatus = useAuthStatus();
+	const isSignedIn =
```

Comment:
I think we'd want to check for ad free rather than signed in no? We still want to show it to signed in users who aren't subscribers if they're using an ad blocker.

Reply:
I copied the logic directly from the old shady pie implementation #3220. I'll get some clarification on the desired behaviour, as I agree we may still want to enable this for signed in users that aren't ad-free!

Reply:
Ah I see, will leave it to you to fix here or in a followup if needed!

Reply:
Thanks for catching this though. I've checked this now and I think we will change the logic. I'll do that as a follow up though, as this is all behind a 0% test anyway.